clients/src/test/java/org/apache/kafka/clients/producer/internals/TransactionManagerTest.java

```diff
@@ -876,7 +876,7 @@ public void testUnsupportedFindCoordinator() {
         client.prepareUnsupportedVersionResponse(body -> {
             FindCoordinatorRequest findCoordinatorRequest = (FindCoordinatorRequest) body;
             assertEquals(CoordinatorType.forId(findCoordinatorRequest.data().keyType()), CoordinatorType.TRANSACTION);
-            assertEquals(findCoordinatorRequest.data().key(), transactionalId);
```

Comment:
The key should be empty as the request is using the batch keys. Also, the collection of `coordinatorKeys` should include only one element (copy from `key`)

Reply:
Updated it. Thank you.